src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -639,6 +645,33 @@ private void DoStart()
             if (logger.IsVerbose) { logger.Verbose("Silo.Start complete: System status = {0}", SystemStatus.Current); }
         }
 
+        private void CreateGrainServices(GrainServiceConfigurations grainServiceConfigurations)
+        {
+            foreach (var serviceConfig in grainServiceConfigurations.GrainServices)
+            {
+                // Construct the Grain Service
+                var serviceType = System.Type.GetType(serviceConfig.Value.ServiceType);
+                if (serviceType == null)
+                {
+                    throw new Exception(String.Format("Cannot find Grain Service type {0} of Grain Service {1}", serviceConfig.Value.ServiceType, serviceConfig.Value.Name));
+                }
+
+                // internal GrainService(GrainId grainId, Silo silo) 
+                var ctor = TypeUtils.GetConstructorThatMatches(serviceType, new[] {typeof(object), typeof(Silo)});
```

Comment:
I know it's late in the process for this, but we ought to be managing construction and activation like we do with Grains. We should support construction via DI and configure the internals before activation.

I'm ok to merge this as-is, but I'd prefer we address this pre-1.4.0